frontend/src/components/AppLogo.vue

```diff
@@ -1,16 +1,57 @@
-<template>
-  <img src="../../public/icons/argo_icona.svg" alt="Logo" height="40" class="logo" />
-</template>
+<script lang="ts">
+// External libraries
+import { computed, defineComponent } from 'vue';
+
+// Internal dependencies
+import { useLayout } from '@/composables/layout';
+
+export default defineComponent({
+  name: 'AppLogo',
+  /**
+   * Props for AppLogo component.
+   * @prop {String} path - The URL or path to the image.
+   * @prop {String} [width='auto'] - The width of the image. Default is 'auto'.
+   * @prop {String} [height='auto'] - The height of the image. Default is 'auto'.
+   */
+  props: {
+    path: {
+      type: String,
+      required: true
+    },
+    width: {
+      type: String,
+      default: 'auto'
+    },
+    height: {
+      type: String,
+      default: 'auto'
+    }
+  },
+  setup(props) {
```

Comment:
```suggestion
  setup() {
```
non usata per rimuovere il warning